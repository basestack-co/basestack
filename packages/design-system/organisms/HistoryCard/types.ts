import { SpaceProps } from "styled-system";

export type Type =
  | "deleted"
  | "edited"
  | "created"
  | "toggledOn"
  | "toggledOff";

export interface HistoryCardProps extends SpaceProps {
  userName: string;
  description: string;
  flagName: string;
  date: string;
  environment: string;
  type: Type;
  hasPaddingTop?: boolean;
  hasPaddingBottom?: boolean;
}
