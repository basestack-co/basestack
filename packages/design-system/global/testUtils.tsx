import { type RenderOptions, render } from "@testing-library/react";
import type React from "react";
// Styles
import { ThemeProvider } from "styled-components";
import theme from "../theme/lightTheme";

interface Props {
  children: React.ReactNode;
}

const WithThemeProvider = ({ children }: Props) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

const renderWithTheme = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "wrapper">,
) => render(ui, { wrapper: WithThemeProvider, ...options });

export { renderWithTheme };
