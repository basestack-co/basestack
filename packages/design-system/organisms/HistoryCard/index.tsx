import React, { memo } from "react";
import { useTheme } from "styled-components";
import { Text, Avatar, Icon } from "../../atoms";
import {
  Container,
  Description,
  IconContainer,
  StyledText,
  TextContainer,
  Wrapper,
} from "./styles";
import { HistoryCardProps } from "./types";

const HistoryCard = ({
  userName,
  description,
  flagName,
  date,
  environment,
  type,
  hasPaddingTop = true,
  hasPaddingBottom = true,
}: HistoryCardProps) => {
  const theme = useTheme();

  const icon = {
    deleted: "delete",
    edited: "edit",
    created: "add",
    toggledOn: "toggle_on",
    toggledOff: "toggle_off",
  };

  return (
    <Container
      hasPaddingTop={hasPaddingTop}
      hasPaddingBottom={hasPaddingBottom}
    >
      <Wrapper type={type}>
        <IconContainer>
          <Icon
            icon={type ? icon[type] : "help"}
            size="medium"
            color={theme.colors.gray600}
          />
        </IconContainer>
        <Avatar
          userName={userName}
          alt={`${userName} profile picture`}
          size="small"
          mr={theme.spacing.s3}
        />
        <TextContainer>
          <Text fontWeight={500}>
            {userName} <Description as="span">{description}</Description>{" "}
            <StyledText type={type} as="span">
              {flagName}
            </StyledText>
          </Text>
          <Text muted>{`${date} - ${environment}`}</Text>
        </TextContainer>
      </Wrapper>
    </Container>
  );
};

export default memo(HistoryCard);
