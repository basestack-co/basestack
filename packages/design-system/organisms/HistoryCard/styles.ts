import styled, { css } from "styled-components";
import { space } from "styled-system";
import { Text } from "../../atoms";
import { Type } from "./types";

export const Container = styled.div<{
  hasPaddingTop: boolean;
  hasPaddingBottom: boolean;
}>`
  display: flex;
  flex-direction: column;
  padding-top: ${({ theme, hasPaddingTop }) =>
    hasPaddingTop ? theme.spacing.s3 : 0};
  padding-bottom: ${({ theme, hasPaddingBottom }) =>
    hasPaddingBottom ? theme.spacing.s3 : 0};
  ${space};
`;

export const Wrapper = styled.div<{ type: Type }>`
  display: flex;
  position: relative;

  ${({ theme, type }) =>
    type !== "created" &&
    css`
      &::before {
        content: "";
        position: absolute;
        top: 45px;
        left: 19px;
        bottom: -19px;
        width: 2px;
        background-color: ${theme.colors.gray100};
      }
    `}
`;

export const IconContainer = styled.div`
  display: inline-flex;
  flex-shrink: 0;
  height: 40px;
  width: 40px;
  border-radius: 50%;
  align-items: center;
  justify-content: center;
  background-color: ${({ theme }) => theme.colors.gray100};
  margin-right: ${({ theme }) => theme.spacing.s5};
`;

export const TextContainer = styled.div`
  display: flex;
  flex-direction: column;
  margin-top: 3px;
`;

export const Description = styled(Text)`
  font-weight: 400;
`;

export const StyledText = styled(Text)<{ type: Type }>`
  ${({ type }) =>
    type === "deleted" &&
    css`
      text-decoration: line-through;
    `}
`;
