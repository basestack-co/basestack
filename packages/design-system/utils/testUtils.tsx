import defaultTheme from "@basestack/design-system/theme/lightTheme";
import isPropValid from "@emotion/is-prop-valid";
import { RenderResult, render } from "@testing-library/react";
import React from "react";
import { StyleSheetManager, ThemeProvider } from "styled-components";

const Provider = ({
  children,
  theme = defaultTheme,
}: {
  children: React.ReactNode;
  theme: any;
}) => (
  <ThemeProvider theme={theme}>
    <StyleSheetManager shouldForwardProp={isPropValid}>
      {children}
    </StyleSheetManager>
  </ThemeProvider>
);

export const renderWithTheme = (
  ui: React.ReactElement,
  options?: any,
): RenderResult => render(ui, { wrapper: Provider, ...options });
