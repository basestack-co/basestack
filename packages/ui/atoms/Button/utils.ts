import { Size, Target, Type, Variant } from "@/components/atoms/Button/types";
import { css, DefaultTheme } from "styled-components";
import { darken, rem, transparentize } from "polished";

export const handleButtonSize = (size?: Size) => {
  switch (size) {
    case "large":
      return css`
        height: ${rem("60px")};
        font-size: ${rem("18px")};
        line-height: ${rem("25px")};
        padding: 0 ${rem("40px")};
      `;
    case "medium":
      return css`
        height: ${rem("54px")};
        font-size: ${rem("16px")};
        line-height: ${rem("22px")};
        padding: 0 ${rem("34px")};
      `;
    case "normal":
      return css`
        height: ${rem("48px")};
        font-size: ${rem("16px")};
        line-height: ${rem("22px")};
        padding: 0 ${rem("30px")};
      `;
    default:
    case "small":
      return css`
        height: ${rem("46px")};
        font-size: ${rem("14px")};
        line-height: ${rem("19px")};
        padding: 0 ${rem("24px")};
      `;
    case "xSmall":
      return css`
        height: ${rem("42px")};
        font-size: ${rem("14px")};
        line-height: ${rem("19px")};
        padding: 0 ${rem("20px")};
      `;
  }
};

export const handleButtonsColors = (theme: DefaultTheme) => ({
  primary: {
    outline: {
      color: theme.colors.primary,
      borderColor: theme.colors.primary,
      backgroundColor: "transparent",
    },
    filled: {
      color: theme.colors.white,
      borderColor: "transparent",
      backgroundColor: theme.colors.primary,
    },
  },
  secondary: {
    outline: {
      color: theme.colors.secondary,
      borderColor: theme.colors.secondary,
      backgroundColor: "transparent",
    },
    filled: {
      color: theme.colors.white,
      borderColor: "transparent",
      backgroundColor: theme.colors.secondary,
    },
  },
  tertiary: {
    outline: {
      color: theme.colors.gray3,
      borderColor: theme.colors.gray3,
      backgroundColor: "transparent",
    },
    filled: {
      color: theme.colors.primary,
      borderColor: "transparent",
      backgroundColor: theme.colors.gray6,
    },
  },
  neutral: {
    outline: {
      color: theme.colors.gray2,
      borderColor: theme.colors.gray2,
      backgroundColor: "transparent",
    },
    filled: {
      color: theme.colors.primary,
      borderColor: "transparent",
      backgroundColor: "transparent",
    },
  },
  success: {
    outline: {
      color: theme.colors.success,
      borderColor: theme.colors.success,
      backgroundColor: "transparent",
    },
    filled: {
      color: theme.colors.white,
      borderColor: "transparent",
      backgroundColor: theme.colors.success,
    },
  },
  warning: {
    outline: {
      color: theme.colors.warning,
      borderColor: theme.colors.warning,
      backgroundColor: "transparent",
    },
    filled: {
      color: theme.colors.white,
      borderColor: "transparent",
      backgroundColor: theme.colors.warning,
    },
  },
  danger: {
    outline: {
      color: theme.colors.error,
      borderColor: theme.colors.error,
      backgroundColor: "transparent",
    },
    filled: {
      color: theme.colors.white,
      borderColor: "transparent",
      backgroundColor: theme.colors.error,
    },
  },
});

export const darkenColors = (
  theme: DefaultTheme,
  amount: number,
  type: Type,
  variant: Variant,
  target: Target
) => {
  return darken(
    amount || 0.1,
    handleButtonsColors(theme)[type || "primary"][variant || "filled"][target]
  );
};

export const lightenColors = (
  theme: DefaultTheme,
  amount: number,
  type: Type,
  variant: Variant,
  target: Target
) => {
  return transparentize(
    amount || 0.6,
    handleButtonsColors(theme)[type || "primary"][variant || "filled"][target]
  );
};
