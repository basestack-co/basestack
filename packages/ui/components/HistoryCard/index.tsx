import React, { memo } from "react";
import { useTheme } from "styled-components";
import { Text, Avatar, Icon, Label } from "@basestack/design-system";
import {
  BottomContentContainer,
  Container,
  IconContainer,
  LabelsContainer,
  TextContainer,
  TitleContainer,
  Wrapper,
} from "./styles";
import { HistoryCardProps } from "./types";

const HistoryCard = ({
  userName,
  description,
  flagName,
  date,
  environments,
  type,
  hasPaddingTop = true,
  hasPaddingBottom = true,
  avatar,
  hasLeftLine = true,
}: HistoryCardProps) => {
  const theme = useTheme();

  const icon = {
    deleted: "delete",
    edited: "edit",
    created: "add",
    createdProject: "folder_open",
    toggledOn: "toggle_on",
    toggledOff: "toggle_off",
  };

  return (
    <Container
      hasPaddingTop={hasPaddingTop}
      hasPaddingBottom={hasPaddingBottom}
    >
      <Wrapper hasLeftLine={hasLeftLine}>
        <IconContainer>
          <Icon
            icon={type ? icon[type] : "help"}
            size="medium"
            color={theme.colors[theme.isDarkMode ? "gray400" : "gray600"]}
          />
        </IconContainer>
        <Avatar
          src={avatar}
          userName={userName}
          alt={`${userName} profile picture`}
          size="small"
          mr={theme.spacing.s3}
        />
        <TextContainer>
          <TitleContainer type={type}>
            <Text flexShrink={0} fontWeight={500}>
              {userName}
            </Text>
            &nbsp;
            <Text flexShrink={0} fontWeight={400}>
              {description}
            </Text>
            &nbsp;
            <Text fontWeight={500}>{flagName}</Text>
          </TitleContainer>
          <BottomContentContainer>
            {environments.length > 0 && (
              <LabelsContainer>
                {environments.map(({ name, enabled }, index) => (
                  <Label
                    key={index}
                    text={name}
                    variant={enabled ? "success" : "default"}
                  />
                ))}
              </LabelsContainer>
            )}
            <Text muted>{date}</Text>
          </BottomContentContainer>
        </TextContainer>
      </Wrapper>
    </Container>
  );
};

export type { HistoryCardProps };

export default memo(HistoryCard);
