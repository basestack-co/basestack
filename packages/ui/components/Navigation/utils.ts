// Types
import { Translate } from "next-translate";
import { NextRouter } from "next/router";
// Utils
import { config } from "@basestack/utils";
import { signOut } from "next-auth/react";

export const getInternalLinks = (t: Translate, formId: string) => {
  return [
    {
      icon: "flag",
      activeText: "Submissions",
      to: `/form/${formId}/submissions`,
      text: t("internal.submissions"),
    },
    {
      icon: "flag",
      activeText: "Setup",
      to: `/form/${formId}/setup`,
      text: t("internal.setup"),
    },
    {
      icon: "settings",
      activeText: "Settings",
      to: `/form/${formId}/settings/general`,
      text: t("internal.settings"),
    },
  ];
};

export const getExternalLinks = (t: Translate) => {
  return [
    {
      icon: "description",
      activeText: "Documentation",
      to: config.urls.docs.flags.base,
      isExternal: true,
      text: t("external.docs"),
    },
    {
      icon: "link",
      activeText: "Github",
      to: config.urls.repo,
      isExternal: true,
      text: t("external.github"),
    },
  ];
};

export const getAvatarDropdownList = (
  t: Translate,
  router: NextRouter,
  onCreateForm: () => void,
) => {
  return [
    {
      icon: "add_circle",
      text: t("create.form"),
      onClick: onCreateForm,
      separator: true,
    },
    {
      icon: "settings",
      text: t("dropdown.settings"),
      onClick: () =>
        router.push({
          pathname: "/user/profile/settings",
        }),
    },
    {
      icon: "logout",
      text: t("dropdown.logout"),
      onClick: signOut,
    },
  ];
};
