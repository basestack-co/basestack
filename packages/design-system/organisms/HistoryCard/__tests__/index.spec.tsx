import { cleanup } from "@testing-library/react";
import { renderWithTheme } from "../../../utils/testUtils";
import HistoryCard from "../index";

describe("HistoryCard Molecule tests", () => {
  afterEach(cleanup);

  test("should render HistoryCard correctly", () => {
    const { asFragment } = renderWithTheme(
      <HistoryCard
        avatar=""
        userName="Joana Lopes"
        description="toggled off"
        flagName="user_image"
        date="1 hour ago"
        environment="Development"
        type="toggledOff"
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
