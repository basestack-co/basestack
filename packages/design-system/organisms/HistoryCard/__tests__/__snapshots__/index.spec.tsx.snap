// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryCard Molecule tests should render HistoryCard correctly 1`] = `
<DocumentFragment>
  .c5 {
  height: 1.75rem;
  width: 1.75rem;
  border-radius: 50%;
  overflow: hidden;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-right: 0.75rem;
  border-radius: 50%;
}

.c6 {
  height: 100%;
  width: 100%;
  object-fit: cover;
}

.c8 {
  font-size: 0.875rem;
  line-height: 1.375rem;
  font-weight: 400;
  font-weight: 500;
  font-family: 'Roboto',sans-serif;
  color: #000000;
}

.c10 {
  font-size: 0.875rem;
  line-height: 1.375rem;
  font-weight: 400;
  font-family: 'Roboto',sans-serif;
  color: #757575;
}

.c3 {
  height: 1.5rem;
  width: 1.5rem;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c4 {
  color: #545454;
  font-size: 1.5rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
}

.c1::before {
  content: "";
  position: absolute;
  top: 45px;
  left: 19px;
  bottom: -19px;
  width: 2px;
  background-color: #EEEEEE;
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  height: 40px;
  width: 40px;
  border-radius: 50%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #EEEEEE;
  margin-right: 1.25rem;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 3px;
}

.c9 {
  font-weight: 400;
}

<div
    class="c0"
  >
    <div
      class="c1"
      type="toggledOff"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
          data-testid="icon-container"
          size="24"
        >
          <span
            class="c4 material-symbols-sharp"
            color="#545454"
            data-testid="icon"
            size="24"
          >
            toggle_off
          </span>
        </div>
      </div>
      <div
        class="c5"
        data-testid="avatar"
        size="28"
      >
        <img
          alt="Joana Lopes profile picture"
          class="c6"
          src="https://avatars.dicebear.com/api/initials/Joana Lopes.svg?size=28&fontSize=40"
        />
      </div>
      <div
        class="c7"
      >
        <p
          class="c8"
          font-weight="500"
        >
          Joana Lopes 
          <span
            class="c9"
          >
            toggled off
          </span>
           
          <span
            class=""
            type="toggledOff"
          >
            user_image
          </span>
        </p>
        <p
          class="c10"
        >
          1 hour ago - Development
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
